### DISCUSSION TITLE
empty Buffalo TS-XL vs.  generate_disk_image.sh

### DISCUSSION THREAD

**Discussion by 4920441** [1000001101000/Debian_on_Buffalo#130] [Q&A]
Hi,

I got this device for free:

https://buffalonas.miraheze.org/wiki/Terastation_TS-XL

Since I don't have any windows box lying around to try this rather crappy looking buffalo software to initialize the buffalo device, I was quiet happy that you made life much more easyier with your script

 generate_disk_image.sh

After adapting the variables  to the TS-XL (I hope)  got an Image....:

`#!/bin/bash

##set some vars.
target="/mnt/target"
target_hostname="buffalobuster"
#target_hostname="ls410d"
machine="Buffalo Nas WXL"
#machine="Buffalo Linkstation LS410D"
#machine="Buffalo Linkstation LS-QVL"
arch="armel"
#arch="armhf"
target_rootpw="secretrootpassword"
target_user="user"
target_userpw="usecretuserpassword"
distro="stretch"
#distro="buster"
#distro="buster"

boot_size="512"
swap_size="1024"
root_size="2048"

use_raid="N"
has_micon="Y"

if [ "$machine" == "Buffalo Terastation Pro II/Live" ] || [ "$machine" == "Buffalo Terastation TS-XEL" ] || [ "$machine" == "Buffalo Nas WXL" ] || [ "$machine" == "Buffalo Terastation III" ] || [ "$machine" == "Buffalo Linkstation Pro/Live" ] || [ "$machine" == "Buffalo/Revogear Kurobox Pro" ]; then
  has_micon="Y"
fi

chroot_only="N"
`

This Image I put on an empty harddrive, doublechecking all   partitions were correctly created (which they are).
But after a couple of seconds the  display shows only "no init file hdd2" - nothing else happened.

I remembered the device might only boot with at least two drives attached, so I put another identical drive in slot 2 and rebooted.
Now I saw, "no init file hdd3".... okay...? but why?

What am I doing wrong? 

Maybe are my variables not okay for this device?

I already tried it with buster, bullseye (which should not work as far as I read the wiki) and even stretch, but everytime I only get No Init file on HDD 3 (or 2)

Thanks a alot for your help!

Cheers
4920441

**Comment by 1000001101000** [comment]
Huh, looks like i’m missing some logic in the script for that device…. how embarassing.

If you look at the bottom you can see where it installs the custom kernel for the terstation II. If you change the machine name in that if statement to match your device it should work properly.

I’ll update the script when I get a moment.

**Reply by 4920441** [reply]
Cool :-)  So I stick to the installer instead:-)

Now I partitioned the hdd and put uImage and initrd on this - currently it is installting, I just wonder if the reboot is successful as well.

Keep you informed:-)
